src/x86/nt.rs

```diff
@@ -0,0 +1,275 @@
+//! Non-temporal store operations.
+use core::{marker::PhantomData, ptr};
+
+#[cfg(target_arch = "x86")]
+use core::arch::x86::{
+    self as arch, __m128, __m128d, __m128i, __m256, __m256d, __m256i, _mm_sfence,
+};
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64::{
+    self as arch, __m128, __m128d, __m128i, __m256, __m256d, __m256i, _mm_sfence,
+};
+
+/// Load from a 32-bit aligned address with non-temporal hint, avoiding filling the cache.
+#[inline]
+#[cfg(any())]
+#[target_feature(enable = "avx2")]
+pub fn _mm256_stream_load_si256(addr: &__m256i) -> __m256i {
+    unsafe { arch::_mm256_stream_load_si256(addr) }
+}
+
+/// Store into a 32-bit aligned address with non-temporal hint, avoiding clobbering the cache.
+#[inline]
+#[target_feature(enable = "avx")]
+pub fn _mm256_stream_store_256i(addr: &mut NonTemporalStoreable<'_, __m256i>, v: __m256i) {
+    unsafe { arch::_mm256_stream_si256(addr.inner.as_ptr(), v) }
+}
```

Comment:
```suggestion
pub fn _mm256_stream_si256(addr: &mut NonTemporalStoreable<'_, __m256i>, v: __m256i) {
    unsafe { arch::_mm256_stream_si256(addr.inner.as_ptr(), v) }
}

/// Alias for [`_mm256_stream_si256`].
#[inline]
#[target_feature(enable = "avx")]
pub fn _mm256_stream_store_si256(addr: &mut NonTemporalStoreable<'_, __m256i>, v: __m256i) {
    _mm256_stream_si256(addr.inner.as_ptr(), v)
}
```

I'd prefer not to deviate from the intrinsic name. 
I do think having a version with `_store_` in the name is justified but I would like it to be an alias for the original intrinsic name.

(There are upstream examples of duplicated/alias functions, such as `_mm_load_ps1`/`_mm_load1_ps`).